### PR TITLE
Delay the IP scan to make sure that no further candidates are added

### PR DESCRIPTION
**Pull request by kaikreuzer**
…and do not use a dedicated thread pool

### CODE REVIEW

File: bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java [hunk 1]
```diff
@@ -81,7 +80,8 @@ public class IpAddonFinder extends BaseAddonFinder {
     private static final String PARAMETER_TIMEOUT_MS = "timeoutMs";
 
     private final Logger logger = LoggerFactory.getLogger(IpAddonFinder.class);
-    private final ScheduledExecutorService scheduler = ThreadPoolManager.getScheduledPool(SERVICE_NAME);
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java:84]: This created a dedicated pool of 5 threads, although there is a single job to be run only a single time.

File: bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java [hunk 2]
```diff
@@ -96,34 +96,30 @@ public void deactivate() {
...
+    @Override
     public void setAddonCandidates(List<AddonInfo> candidates) {
         logger.debug("IpAddonFinder::setAddonCandidates({})", candidates.size());
         super.setAddonCandidates(candidates);
         startScan();
     }
 
-    synchronized void startScan() {
-        if (scanJob == null) {
-            scanJob = scheduler.schedule(this::scan, 1, TimeUnit.SECONDS);
```
Comment (kaikreuzer) [inline bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java:107]: The `setAddonCandidates` method is called for each info provider.
In order to do the scan only once, but on the full set of candidates, we have to delay the execution.
At the same time we must make sure that a scheduled scan is rescheduled - or (after more than our delay) is executed once more.

Comment (holgerfriedrich) [reply bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java:107]: interesting....
I wonder if somebody could reach the setup wizard before the scan is done..... 20s might be too long.

**Review by mherwege — APPROVED** [review]
LGTM

File: bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java (hunk 2 above)
Comment (kaikreuzer) [reply bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java:107]: > 20s might be too long

I think it is a good compromise. I doubt that people rush through the wizard. Creating an account and choosing a good password should take them already a while. And if you consider that startup times on a RPi are around 5 minutes, I think we should be on the safe side.

Comment (kaikreuzer) [reply bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java:107]: > btw: maybe we should add your explanation as comment to the code

Done.

**Review by holgerfriedrich — APPROVED** [review]
LGTM

**Comment by kaikreuzer** [comment]
@J-N-K I'll self merge this now as it has 2 approvals and I want to get it into the release.

**Review by J-N-K — COMMENTED** [review]
LGTM

**Pull request merged**